scripts/proteinInteractionMINT/parse_mint.py

```diff
@@ -0,0 +1,418 @@
+    elif source == 'mint':
+        property_line = 'mintID: ' + '"' + id_content + '"'
+    elif source == 'imex':
+        property_line = 'imexID: ' + '"' + id_content + '"'
+    elif source == 'emdb':
+        property_line = 'electronMicroscopyDataBankID: ' + '"' + id_content + '"'
+    elif source == 'wwpdb':
+        property_line = 'worldWideProteinDataBankID: ' + '"' + id_content + '"'
+    elif source == 'rcsb pdb':
+        property_line = 'rcsbPDBID: ' + '"' + id_content + '"'
+    elif source == 'psi-mi':
+        property_line = 'psimiID: ' + '"' + id_content[1:-1] + '"'
+    elif source == 'reactome':
+        property_line = 'reactomePathwayID: ' + '"' + id_content + '"'
+    elif source == 'pdbe':
+        property_line = 'proteinDataBankInEuropeID: '  + '"' + id_content + '"'
+    else:
+        new_source_map[source] = id_content
+        property_line = None
+    return (property_line, new_source_map)
+
+def get_confidence(term):
+    """Convert confidence string to the corresponding confidence property schema
+
+    Args:
+        term: a string with the format "source:id_content"
+
+    Returns:
+        a tuple: (property_line,new_source_map). property_line is the identifier
+        property in schema, and new_source_map is the dictionary of with source
+        name as the key and the identifier as the value, if new source exists.
+        For example:
+
+        ("[13 dcs:AuthorScore]", {aNewSource:10})
+    """
+    source = term.split(':')[0]
+    id_content = ':'.join(term.split(':')[1:])
+    new_source_map = {}
+    if source == 'author score':
+        if id_content.split(" ")[0] == 'Below':
+            property_line = '[- '+ id_content.split(' ')[1] + ' dcs:AuthorScore' +  ']'
+        elif id_content.split(" ")[0] == 'Above':
+            property_line = '['+ id_content.split(' ')[1] + ' - dcs:AuthorScore' +  ']'
+        else:
+            flag = 1
+            # check if author score is a number
+            for part in id_content.split('.'):
+                if not part.isnumeric():
+                    # if author score is "++++"
+                    property_line = '['+ str(len(id_content)) + ' dcs:AuthorScore' +  ']'
+                    flag = 0
+                    break
+            if flag:
+                property_line = '['+ id_content + ' dcs:AuthorScore' +  ']'
+    elif source == 'intact-miscore':
+        property_line = '['+ id_content + ' dcs:IntactMiScore' +  ']'
+    else:
+        new_source_map[source] = id_content
+        property_line = None
+    return (property_line, new_source_map)
+
+def get_protein_dcid(mint_aliases):
+    """Takes a string from the mint database, return the dcid of the protein.
+    Args:
+        mint_aliases: a line contains the aliases of the protein. The capitalized
+        "display_long" name is the dcid of the participant protein.
+    """
+    if len(mint_aliases) > 1:
+        return mint_aliases.split('|')[0].split(':')[1].split('(')[0].upper()
```

Comment:
If the first '|' split didn't have a ':', this function will throw exception. If that can happen, then we should have stricter checks.

Also, consider adding a comment with example.

Reply:
Example added. Here each split is by default to have the ':' in it, with the pattern like this: psi-mi:rpn3_yeast(display_long).